tools/migration/internal/repo_migration_helper.go

```diff
@@ -0,0 +1,64 @@
+package internal
+
+import (
+	"os"
+	"strings"
+
+	"github.com/filecoin-project/go-filecoin/repo"
+)
+
+type RepoMigrationHelper struct {
+	oldRepoPath, newRepoPath, oldVersion, newVersion string
+}
+
+// NewRepoMigrationHelper takes options for old and new repo paths, figures out
+// what the correct paths should be, and creates a new RepoMigrationHelper with the
+// correct paths.
+func NewRepoMigrationHelper(oldRepoOpt, newRepoOpt, oldVersion, newVersion string) *RepoMigrationHelper {
+	oldPath := getOldRepoPath(oldRepoOpt)
+	return &RepoMigrationHelper{
+		newRepoPath: getNewRepoPath(oldPath, newRepoOpt, oldVersion, newVersion),
+		newVersion:  newVersion,
+		oldRepoPath: oldPath,
+		oldVersion:  oldVersion,
+	}
+}
+
+// GetOldRepo returns the old repo dir, opened as read-only.
+func (rmh *RepoMigrationHelper) GetOldRepo() (*os.File, error) {
+	return os.Open(getOldRepoPath(rmh.oldRepoPath))
+}
+
+// MakeNewRepo creates the new repo dir and returns it with Read/Write access.
+func (rmh *RepoMigrationHelper) MakeNewRepo() (*os.File, error) {
+	if err := os.Mkdir(rmh.newRepoPath, os.ModeDir|0744); err != nil {
+		return nil, err
+	}
+	return os.Open(rmh.newRepoPath)
+}
+
+// getOldRepoPath takes a command line option (which can be blank) and uses it
+// to find the correct old repo path.
+func getOldRepoPath(cliOpt string) string {
+	dirname := repo.GetRepoDir(cliOpt)
+	return ExpandHomedir(dirname)
+}
+
+// getNewRepoPath generates a new repo path for a migration.
+// Params:
+//     oldPath:  the actual old repo path
+//  newRepoOpt:  whatever was passed in by the CLI (can be blank)
+//  oldVersion:  old repo version
+//  newVersion:  version to migrate to
+// Returns:  a path generated using all of the above information plus a timestamp.
+// Example output:   /Users/davonte/.filecoin_1_2_2019-08-06_150455
+func getNewRepoPath(oldPath, newRepoOpt, oldVersion, newVersion string) string {
+	var newRepoDir string
+	if newRepoOpt != "" {
+		newRepoDir = newRepoOpt
+	} else {
+		newRepoDir = oldPath
```

Comment:
I think that this nests the new repo within the old one which seems like a problem.

I think it is ok to leave in a bad state until after #2524 lands because that work will make this work out as intended.

Reply:
Note that work in #2524 will break the assumption that the repo to be migrated will be at the path returned by `getOldRepoPath` as written, we'll need to add on a `/repo` somewhere either directly in `getOldRepoPath` or in a new function of the `go-fileocin/repo` module